### PR TITLE
[FLINK-28780][docs] fix incorrect description of function dayofmonth.

### PR DESCRIPTION
**Pull request by liuyongvs**
## What is the purpose of the change

*fix incorrect description of function dayofmonth*

### CODE REVIEW

File: docs/data/sql_functions.yml [hunk 1]
```diff
@@ -463,7 +463,7 @@ temporal:
   - sql: DAYOFYEAR(date)
     description:  Returns the day of a year (an integer between 1 and 366) from SQL date date. Equivalent to EXTRACT(DOY FROM date). E.g., DAYOFYEAR(DATE '1994-09-27') returns 270.
   - sql: DAYOFMONTH
-    description: Returns the day of a month (an integer between 1 and 31) from SQL date date. Equivalent to EXTRACT(DAY FROM date). E.g., DAYOFWEEK(DATE '1994-09-27') returns 3.
+    description: Returns the day of a month (an integer between 1 and 31) from SQL date date. Equivalent to EXTRACT(DAY FROM date). E.g., DAYOFMONTH(DATE '1994-09-27') returns 27.
```
Comment (snuyanzin) [inline docs/data/sql_functions.yml:466]:
```suggestion
    description: Returns the day of a month (an integer between 1 and 31) from SQL date. Equivalent to EXTRACT(DAY FROM date). E.g., DAYOFMONTH(DATE '1994-09-27') returns 27.
```
probably `date date` also should be fixed

Comment (liuyongvs) [reply docs/data/sql_functions.yml]: hi @snuyanzin have done, please review it again thanks.

**Review by snuyanzin — APPROVED** [review]

**Comment by liuyongvs** [comment]
hi @snuyanzin ci passed

**Comment by snuyanzin** [comment]
lgtm from my side however i'm not a committer => i can't merge

**Review by snuyanzin — APPROVED** [review]

**Review by wuchong — APPROVED** [review]
LGTM.

**Comment by wuchong** [comment]
@snuyanzin is right. Would be better to follow the description template especially when the PR is not a simple hotfix.

**Pull request merged**